src/System.Management.Automation/engine/MshObject.cs

```diff
@@ -1269,23 +1269,19 @@ internal static string ToStringParser(ExecutionContext context, object obj, IFor
         /// </exception>
         internal static string ToString(ExecutionContext context, object obj, string separator, string format, IFormatProvider formatProvider, bool recurse, bool unravelEnumeratorOnRecurse)
         {
-            PSObject mshObj = obj as PSObject;
-
-            #region plain object
-            if (mshObj == null)
+            bool TryFastTrackPrimitiveTypes(object obj, out string str)
             {
-                if (obj == null)
-                {
-                    return string.Empty;
-                }
+                str = null;
```

Comment:
I'm a bit averse to having the defaults be to return `true` with `str = null`, and that also loses the nice compile-time check that `str` was set properly.

I think the compiler is smart enough to move `str = null` to the `default` case and still allow the program, while ensuring that any further changes won't return `true` with a null string.

Reply:
Updated.